app/controllers/inventories_controller.rb

```diff
@@ -0,0 +1,77 @@
+class InventoriesController < ApplicationController
+  load_and_authorize_resource
+  before_action :set_inventory, only: %i[show destroy]
+
+  # GET /inventories
+  def index
+    @inventories = current_user.inventories
+  end
+
+  # GET /inventories/1
+  def show
+    @inventory = Inventory.includes(:inventory_foods).find(params[:id])
+  end
+
+  def shopping_list
+    @inventory = Inventory.includes(:inventory_foods).find(params[:inventory_id])
+    @recipe = Recipe.includes(:recipe_foods).find(params[:recipe_id])
+
+    @items = []
+    @total_amount = 0
+    @recipe.recipe_foods.find_each do |recipe_food|
+      inventory_food = @inventory.inventory_foods.find_by(food: recipe_food.food)
+      difference = inventory_food.quantity - recipe_food.quantity
+      if difference.negative?
+        food = recipe_food.food
+        quantity = difference.abs
+        price = quantity * food.price
+        name = food.name
+        measurement_unit = food.measurement_unit
+
+        food_item = Struct.new(:name, :price, :quantity, :measurement_unit)
+        item = food_item.new(name, price, quantity, measurement_unit)
+        @total_amount += price
+        @items.push(item)
+      end
+    end
+    @items_count = @items.length
+  end
```

Comment:
- [ ] The controllers should be concerned with HTTP verbs. (thin controller rule) By that, I mean that a controller’s responsibility should be focused on web requests. Other calculations should be handled on the model.